### ISSUE TITLE
SYCL error in compilation stage

### ISSUE THREAD

**Issue by ruialves7** [UoB-HPC/BabelStream#117]
Hello,
When I try to compile to the sycl model using computecpp, I get the following error: BabelStream/src/sycl/SYCLStream.h:37:5: error: use of undeclared identifier 'cl'

I've already tried a number of symlinks but I couldn't find a solution

Regards, Rui

**Comment by tomdeakin** [comment]
Which version of ComputeCpp are you using?

**Comment by ruialves7** [comment]
2.8.0, more exactly the last version.

Thanks 

Rui

**Comment by tomdeakin** [comment]
The following works for me on CentOS 7.6 with a local install of GCC 9.1 and the Khronos OpenCL ICD with ComputeCpp 2.8.0:

```bash
cmake -DMODEL=sycl -DSYCL_COMPILER=COMPUTECPP -DSYCL_COMPILER_DIR=/path/to/ComputeCpp-CE-2.8.0-x86_64-linux-gnu/ -DOpenCL_LIBRARY=/path/to/icd/libOpenCL.so -Bbuild -H.
```

Note, I made sure to remove the `babelstream/build` directory first to ensure no CMake conflicts.

If you are still having trouble, please share details of your system along with the exact build command you are using, along with the error you see.

Hopefully the example above should help guide your build.

**Comment by ruialves7** [comment]
Hi, 

I execute the same command and do your suggestion, but is not working ;(.
My specs: gcc 6.3.0,Ubuntu 20.04.3 LTS, Khronos OpenCL ICD.

should i try other version of the computeCPP?

**Comment by tomdeakin** [comment]
Can you copy/paste your exact `cmake` command, and the entire output from the build process please?

**Comment by ruialves7** [comment]
Hi, 

My cmake command:

cmake -Bbuild -H. -DMODEL=sycl -D  SYCL_COMPILER=COMPUTECPP -D SYCL_COMPILER_DIR=/opt/ComputeCPP -D OpenCL_LIBRARY=/usr/lib/usr/lib/x86_64-linux-gnu/libOpenCL.so -Bbuild -H


My Output error:
In file included from /home/ropencl-user/babel/BabelStream/src/sycl/SYCLStream.cpp:8:
/home/ropencl-user/babel/BabelStream/src/sycl/SYCLStream.h:35:5: error: use of
      undeclared identifier 'cl'
    cl::sycl::queue *queue;
    ^
/home/ropencl-user/babel/BabelStream/src/sycl/SYCLStream.h:36:5: error: use of
      undeclared identifier 'cl'
    cl::sycl::buffer<T> *d_a;

The local OpenCL installation works as it has already run other benchmarks.

Thank you, 

Rui

**Comment by tomdeakin** [comment]
Any ideas on this one @Ruyk?

**Comment by Ruyk** [comment]
Never seen that before. I've just used the same cmake line above, minus the extra "-H" and built fine on my desktop.

**Comment by ruialves7** [comment]
Hi Thank you for your help. I suspect that the ComputeCPP version, even though it is the most recent, has issues. When I tried to compile the first time I got an error ( nested include). I'll try it with another version and then share the result. Thanks a lot for the help.

**Comment by tom91136** [comment]
Could you upload the full output of the following two commands, make sure you capture stderr and stdout:

     # rm -rf build # remove build first
     cmake -H. -Bbuild -DMODEL=sycl <your flags here...>  -DCMAKE_VERBOSE_MAKEFILE=ON -DCXX_EXTRA_FLAGS=-v 
     cmake --build build 

The output will be very long so just drag the `txt` file here.

**Comment by ruialves7** [comment]
Hi,

I compiled the sycl model using ComputeCPP 2.0.0. With ComputeCPP 2.8.0 I couldn't , it gives a nested include error. I downloaded the files from the ComputeCPP website, and if I check the ComputeCPP/include/CL/sycl.hpp directory and the ComputeCPP/include/sycl/sycl.hpp directory, it seems to me that the content is strange.

after this, i compiled and tested what i wanted

 Thank you for your help

Rui
